server/v2/core/stf/stf.go

```diff
@@ -17,12 +19,39 @@ type STF[T transaction.Tx] interface {
 	DeliverBlock(
 		ctx context.Context,
 		block *appmanager.BlockRequest[T],
-		state store.GetReader,
-	) (*appmanager.BlockResponse, store.GetWriter, error)
+		state store.ReaderMap,
+	) (*appmanager.BlockResponse, store.WriterMap, error)
 	// Simulate simulates the execution of a transaction over the provided state, with the provided gas limit.
-	Simulate(ctx context.Context, state store.GetReader, gasLimit uint64, tx T) (appmanager.TxResult, store.GetWriter)
+	Simulate(ctx context.Context, state store.ReaderMap, gasLimit uint64, tx T) (appmanager.TxResult, store.WriterMap)
 	// Query runs the provided query over the provided readonly state.
-	Query(ctx context.Context, state store.GetReader, gasLimit uint64, queryRequest appmanager.Type) (queryResponse appmanager.Type, err error)
+	Query(ctx context.Context, state store.ReaderMap, gasLimit uint64, queryRequest appmanager.Type) (queryResponse appmanager.Type, err error)
 	// ValidateTx validates the TX.
-	ValidateTx(ctx context.Context, state store.GetReader, gasLimit uint64, tx T) appmanager.TxResult
+	ValidateTx(ctx context.Context, state store.ReaderMap, gasLimit uint64, tx T) appmanager.TxResult
+}
+
+// ErrOutOfGas must be used by GasMeter implementers to signal
+// that the state transition consumed all the allowed computational
+// gas.
+var ErrOutOfGas = errors.New("out of gas")
+
+// Gas defines type alias of uint64 for gas consumption. Gas is used
+// to measure computational overhead when executing state transitions,
+// it might be related to storage access and not only.
+type Gas = uint64
+
+// NoGasLimit signals that no gas limit must be applied.
+const NoGasLimit Gas = math.MaxUint64
+
+// GasMeter defines an interface for gas consumption tracking.
+type GasMeter interface {
```

Comment:
we should try to merge this with the one in cosmossdk.io/core

Reply:
agreed

Reply:
i think when we re-organize server/core we can decide if STF imports core/gas or if it expects an extension of core/gas